msal-device-flow-python/app.py

```diff
@@ -0,0 +1,38 @@
+"""
+A Python application that demonstrates how to use the
+Device Code flow to make an API call to Microsoft Graph.
+"""
+
+# Microsoft Authentication Library (MSAL) for Python
+import msal
+
+# Used to allow the app to make the HTTP request to Graph
+import requests
+
+# MSAL configs
+config = {
+    # Full directory URL, in the form of https://login.microsoftonline.com/<tenant>
+	"authority": "",
+
+    # 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+	"client_id": "",
+
+    # The User.Read scope is needed for the Graph API request
+	"scopes": ["User.Read"]
+}
+
+# Create connection to API via MSAL
+msalClientApp = msal.PublicClientApplication(config["client_id"], authority=config["authority"])
+
+# Initialize the Divice Code flow for the desired scope(s)
+deviceFlow = msalClientApp.initiate_device_flow(scopes=config["scopes"])
+
+# Displays a message instructing the user to authenticate via their browser
+print(deviceFlow["message"], flush=True)
+
+# Get the access token from MSAL
+msalResponse = msalClientApp.acquire_token_by_device_flow(deviceFlow)
```

Comment:
Does this "pause" here until the process completes?  If the user doesn't complete or completes with failure, can we `print` out a message to handle that?

Reply:
It does pause/resume upon success.  Will see what happens with failure.  I tried it once and it just continued sitting there.

Reply:
One way to test would be to just cancel the login flow.

Reply:
@ckittel Trying to think what would be ideal here.  The present way, it will wait forever until the user completes successfully or terminates the program.  The alternative would be to perform polling with this function-> acquire_token_by_device_flow

Reply:
waiting forever is fine, just need to handle what happens when the user actively cancels the flow (like they had a typo in their username or they selected the wrong one).  What is the "user initiated" cancel experience?

Reply:
If I understand the documentation correctly, there is no way for the user to initiate a cancel experience.   The only option appears to be a timeout (vs waiting literally, forever)

Reply:
Thanks, good to know.